### ISSUE TITLE
`Slightly` different training loss and detection eval in different training

### ISSUE THREAD

**Issue by stoneyang** [weiliu89/caffe#173]
Hi, @weiliu89 

Thanks for your cool work and excellent code! 

I am trying to reproduce your results strictly according the instructions provided in [README](https://github.com/weiliu89/caffe/tree/ssd#ssd-single-shot-multibox-detector). 

But something seems weird to me: for different training using same `ssd_pascal.py` running on the same GPU (one GTX 1070, only changing [`batch_size`](https://github.com/weiliu89/caffe/blob/ssd/examples/ssd/ssd_pascal.py#L290) to `8` in order to fit the input to the memory), the training threw out different losses and detection evals. 

1.  detection_eval: 0.720876
2.  detection_eval: 0.719783
3.  detection_eval: 0.722221
4.  detection_eval: 0.720411

Although they are at some approximation of 72.1% as your paper noted, could these training on my machine make sense? I mean, is there something wrong in my training? Otherwise, could these slightly fluctuation be tolerated? And, what introduces the fluctuation? SGD, Xavier filler, precision of HW, shuffling input?

Any suggestion would be appreciated. 
Thanks in advance!

**Comment by weiliu89** [comment]
Because the "new" layers' parameters are initialized randomly, so it is normal that the mAP is slightly different for each run.

**Comment by stoneyang** [comment]
Timely reply! 

So, at this setting, nothing to do with SGD, HW precision, or shuffling input?

**Comment by weiliu89** [comment]
Besides there is data augmentation during training, which is random as well. I don't think it has anything to do with HW precision. It doesn't shuffle input during training.

**Comment by stoneyang** [comment]
Got it, thanks a lot!
